### PR TITLE
feat(shell-api): add autoMerger helpers MONGOSH-1375

### PR DESCRIPTION
**Pull request by lerouxb**

### CODE REVIEW

File: packages/shell-api/src/shard.ts [hunk 1]
```diff
@@ -464,4 +464,73 @@ export default class Shard extends ShellApiWithMongoClass {
...
+      { _id: 'automerge' },
+      { $set: { enabled: false } },
+      { upsert: true, writeConcern: { w: 'majority', wtimeout: 30000 } }
+    ) as UpdateResult;
+  }
+
+  @returnsPromise
+  @apiVersions([])
+  @serverVersions(['7.0.0', ServerVersions.latest])
+  async shouldAutoMerge(): Promise<boolean> {
```
Comment (addaleax) [inline packages/shell-api/src/shard.ts:497]: Did the sharding team come up with this name? I’m wondering if something like “sh.isAutoMergerEnabled()” is a better name, also considering that “sh.isBalancerRunning()” is a thing

File: packages/shell-api/src/shard.spec.ts [hunk 2]
```diff
@@ -1217,6 +1217,275 @@ describe('Shard', () => {
...
+      });
+
+      it('throws if not mongos', async() => {
+        const expectedResult = { ok: 1 } as any;
+        serviceProvider.runCommandWithCheck.resolves({ ok: 1, msg: 'not dbgrid' });
+        serviceProvider.updateOne.resolves(expectedResult);
+        await shard.enableAutoMerger('ns');
+        expect(warnSpy.calledOnce).to.equal(true);
+      });
+    });
```
Comment (addaleax) [inline packages/shell-api/src/shard.spec.ts:1488]: To be clear, what I meant on Slack was that I had assumed that there were tests in the `integration` section below :)

Comment (lerouxb) [reply packages/shell-api/src/shard.spec.ts:1488]: Will check. I thought that I copied all the code for the things I copy/pasted from :)

File: packages/shell-api/src/shard.ts (hunk 1 above)
Comment (lerouxb) [reply packages/shell-api/src/shard.ts:497]: Yes I copied from where the ticket said I should copy from: https://github.com/10gen/mongo/pull/11291/files

Comment (addaleax) [reply packages/shell-api/src/shard.ts:497]: Yeah, that’s the nice thing about the legacy shell not shipping anymore … we can do whatever we think is best in terms of UX.

**Review by addaleax — APPROVED** [review]

**Comment by lerouxb** [comment]
I think these errors are all things we're seeing elsewhere, so should be safe to merge.

**Pull request merged**